src/wrapper/objects/jinputstream.rs

```diff
@@ -0,0 +1,86 @@
+use crate::{
+    errors::*,
+    objects::{AutoLocal, JMethodID, JObject},
+    signature::{JavaType, Primitive},
+    sys::jsize,
+    JNIEnv,
+};
+use std::io;
+
+const DEFAULT_BUF_SIZE: usize = 8 * 1024; // same as std::io
+
+/// Wrapper for JObjects that implement `java.io.InputStream`.
+pub struct JInputStream<'a: 'b, 'b> {
+    internal: JObject<'a>,
+    buffer: AutoLocal<'a, 'b>,
+    read: JMethodID<'a>,
+    env: &'b JNIEnv<'a>,
```

Comment:
JNIEnv is actually clonable. Is there a reason to use reference here?

Reply:
Actually I just copied and pasted `jmap.rs` to begin with 😄, and this is the way it's done there. I agree that there's nothing to be gained by taking a reference since it's just a pointer internally, but I suppose it's good to do it consistently in all the wrappers?

Reply:
I think `jmap.rs` was there before `JNIEnv` was made clonable :)

It looks you will still need second lifetime parameter, so we can leave it as it is.